### ISSUE TITLE
Package request: hoppscotch

### ISSUE THREAD

**Issue by felixsanz** [NixOS/nixpkgs#261872]
**Project description**

Open source API development ecosystem - https://hoppscotch.io/

It's a web service for API testing, like postman/insomnia but self-hosted and doesn't run on electron. Uses postgresql.

Demo: https://hoppscotch.io/

**Metadata**

* homepage URL: https://github.com/hoppscotch/hoppscotch/
* source URL: https://github.com/hoppscotch/hoppscotch/
* dockerfile: https://raw.githubusercontent.com/hoppscotch/hoppscotch/main/prod.Dockerfile
* docker compose file: https://raw.githubusercontent.com/hoppscotch/hoppscotch/main/docker-compose.yml
* license: MIT
* platforms: unix, linux, darwin, ...

**Comment by humancalico** [comment]
They're just released a desktop app https://hoppscotch.com/download

**Comment by GetPsyched** [comment]
Hey, @felixsanz, from what I've discussed with the upstream devs, this will most likely be possible only once their new desktop app (as @humancalico mentioned) is decoupled from the code that connects then to the `hoppscotch.io` servers.

So, essentially, they'll allow a configuration option in the future between self hosting, hoppscotch servers, etc. either through multiple versions of their app, or via in-app options (this is TBD since their desktop app is still in alpha)

**Comment by felixsanz** [comment]
> Hey, @felixsanz, from what I've discussed with the upstream devs, this will most likely be possible only once their new desktop app (as @humancalico mentioned) is decoupled from the code that connects the app to the `hoppscotch.io` servers.
> 
> So, essentially, they'll allow a configuration option in the future between self hosting, hoppscotch servers, etc. either through multiple versions of their app, or via in-app options (this is TBD since their desktop app is still in alpha)

Just to note, i don't know others but i wouldn't be interested in another electron app. i would only use hoppscotch if it's available as a self-host service, which is how it runs now

**Comment by GetPsyched** [comment]
I'll discuss this with the devs; currently, I have no idea how something like that would be packaged.

**Comment by felixsanz** [comment]
> I'll discuss this with the devs; currently, I have no idea how something like that would be packaged.

just like miniflux is packaged. it's a service that runs via systemd and available as a self-hosted web app

**Comment by GetPsyched** [comment]
> > I'll discuss this with the devs; currently, I have no idea how something like that would be packaged.
> 
> just like miniflux is packaged. it's a service that runs via systemd and available as a self-hosted web app.
> 
> https://raw.githubusercontent.com/hoppscotch/hoppscotch/main/docker-compose.yml

I'll look into it; although I'll get quite busy soon so no promises on this from my end. Any maintainer may feel free to package it themselves early.

**Comment by GetPsyched** [comment]
Yes, that's the desktop app; this issue specifically requests the self-hosted module.

**Comment by DataHearth** [comment]
Hey! I've already packaged the desktop app in #289271 😄 . Tested on macos and linux (x64 arch).

**Comment by GetPsyched** [comment]
Hey, please read the above discussion and the original post; this issue requests the self hosted runner of Hoppscotch and *not* the desktop application. Thanks.
